packages/neon-core/src/sc/ScriptReader.ts

```diff
@@ -0,0 +1,121 @@
+import { reverseHex, StringStream, hexstring2str } from "../u";
+import { ScriptIntent } from "./ScriptBuilder";
+import InteropServiceCode from "./InteropServiceCode";
+
+function checkArgs(args?: any[]): void {
+  if (
+    !args ||
+    args.length !== 3 ||
+    !Array.isArray(args[2]) ||
+    typeof args[1] !== "string" ||
+    typeof args[0] !== "string"
+  ) {
+    throw new Error(`Script Intent Not In the right format: ${args}`);
+  }
+}
+
+/**
+ * Read a script to readable contract invocations.
+ * @extends StringStream
+ */
+export class ScriptReader extends StringStream {
```

Comment:
ScriptParser would be a better name.

Reply:
OK. I will relieve `ScriptBuilder` of extending `StringStream`, since it is not an string reader anymore, OK?

Reply:
I would prefer that you stop adding more ideas to this PR and focus on just getting this closed.